examples/albert/schedule.py

```diff
@@ -232,24 +234,24 @@ def shard_word_embedding(sch, vocab_size, word_embed_name="embeddings.word_embed
     vocab_start_index = sch.rank * vocab_size // sch.world_size
     vocab_end_index = (sch.rank + 1) * vocab_size // sch.world_size
 
-    def fw_pre_hook(_input):
+    def fwd_pre_hook(_module, _input):
         # Mask the input
         input_mask = (_input[0] < vocab_start_index) | (_input[0] >= vocab_end_index)
```

Comment:
btw, _input here has shape (B, T), so we should just check `_input < vocab_start_index) | (_input >= vocab_end_index)`, otherwise `_input[0] is only for the first example`?

Reply:
IIUC this input is always a list, so input[0] should be the entire tensor.

Reply:
wait, if this is a list, why we only return the masked_input? the module should expect the same input structure?

Reply:
That's because this case only has one input tensor. In this case pytorch will wrap a single returned tensor to a tuple, according to https://pytorch.org/docs/stable/generated/torch.nn.modules.module.register_module_forward_pre_hook.html

Reply:
ok makes sense